src/api/tests/customers/getAll.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { customersAllSchema } from "data/schemas/customers/customersAll.schema";
+import { STATUS_CODES } from "data/statusCodes";
+import { validateSchema } from "utils/validations/schemaValidation";
+
+//smoke test без учета фильтрационных параметров
+test.describe("[API] [Customers] [Get All Customers ]", () => {
+  test("Should get All Customers without params", async ({ request }) => {
+    //login
+    const loginResponse = await request.post(apiConfig.BASE_URL + apiConfig.ENDPOINTS.LOGIN, {
+      data: { username: USER_LOGIN, password: USER_PASSWORD },
+      headers: {
+        "content-type": "application/json",
+      },
+    });
+    const headersResponse = loginResponse.headers();
+    const authToken = headersResponse["authorization"];
+    expect.soft(loginResponse.status()).toBe(STATUS_CODES.OK);
+
+    //создание нового customer
+    const customerData = generateCustomerData();
+    const customerResponse = await request.post(apiConfig.BASE_URL + apiConfig.ENDPOINTS.CUSTOMERS, {
+      data: customerData,
+      headers: {
+        "content-type": "application/json",
+        Authorization: `Bearer ${authToken}`,
+      },
+    });
+    //проверка создания customer
+    const customerBody = await customerResponse.json();
+    expect.soft(customerResponse.status()).toBe(STATUS_CODES.CREATED);
+
+    //отправка  GET запроса для получения всех customers(без парам)
+    const getResponse = await request.get(apiConfig.BASE_URL + apiConfig.ENDPOINTS.CUSTOMERS, {
+      headers: {
+        Authorization: `Bearer ${authToken}`,
+        "content-type": "application/json",
+      },
+    });
+
+    //asserts
+    const bodyResponseAllCustomers = await getResponse.json();
+    // Валидация схемы ответа
+    validateSchema(customersAllSchema, bodyResponseAllCustomers);
+    // Проверка статуса GET запроса
+    expect.soft(getResponse.status()).toBe(STATUS_CODES.OK);
+    // Проверка тела ответа (наличие созданного customer in arr)
+    //1var
+    // проверяем ч/з some(есть ли хотя бы один true ) что есть хоть  один  customer-obj с ключом id, email,  name (выбрала основные поля)
+    const isCustomerPresentId = bodyResponseAllCustomers.Customers.some((customer: { _id: string }) => customer._id === customerBody.Customer._id);
+    expect.soft(isCustomerPresentId).toBe(true);
+    const isCustomerPresentEmail = bodyResponseAllCustomers.Customers.some((customer: { email: string }) => customer.email === customerBody.Customer.email);
+    expect.soft(isCustomerPresentEmail).toBe(true);
+    const isCustomerPresentName = bodyResponseAllCustomers.Customers.some((customer: { name: string }) => customer.name === customerBody.Customer.name);
+    expect.soft(isCustomerPresentName).toBe(true);
+    //2var
+    const newCustomer = bodyResponseAllCustomers.Customers.find((customer: { email: string }) => customer.email === customerData.email);
+    expect.soft(newCustomer).toMatchObject({ ...customerBody.Customer });
```

Comment:
Сверять скорее нужно с данными которые мы отправляли, а не получили от сервера, мы не знаем мутировали они или нет

```suggestion
    expect.soft(newCustomer).toMatchObject(customerData);
```

Reply:
```
типо такого наверно?
    const newCustomer = bodyResponseAllCustomers.Customers.find((customer: { _id: string }) => customer._id === customerBody.Customer._id);
    const { _id, createdOn, ...rest } = newCustomer;
    expect.soft(rest).toMatchObject({
      email: customerData.email,
      name: customerData.name,
      city: customerData.city,
      country: customerData.country,
      flat: customerData.flat,
      house: customerData.house,
      notes: customerData.notes,
    });

```

Reply:
https://jestjs.io/docs/expect#tomatchobjectobject
`.toMatchObject is called for each elements, so extra object properties are okay`

В нашем случае получается можно не делать деструктуризацию и сравнивать в таком порядке кажется

`expect.soft(newCustomer).toMatchObject(customerData);`

Reply:
Обновила с разными вариантами:  `toMatchObject(customerData)` высвечивает ошибку, только через явное указание полей сработало. 
добавила вариант еще ч/з .toEqual(customerData) и деструктуризацию

Reply:
Надо разобраться в ошибке, все работает
```ts

    const customerData = generateNewCustomer();
    await CustomersController.create({ data: customerData, token });
    const customersResponse = await CustomersController.getAll({ token });

    const createdCustomer = customersResponse.data.Customers.find((c: ICustomerFromResponse) =>
      c.email === customerData.email,
    );

    expect(createdCustomer,
      'Customer from response does not match created customer data')
      .toMatchObject({ ...customerData }); // ругается на типизацию, если передавать просто объект
```

Reply:
<img width="1075" alt="image" src="https://github.com/user-attachments/assets/b34d6425-fe18-47f6-bf7f-0821bab0985b" />

Reply:
Все верно, у меня так же
Я в примере рассписал в комменте после toMatchObject